### DISCUSSION TITLE
how to run nx dev and dependencies watch together

### DISCUSSION THREAD

**Discussion by khanakia** [nrwl/nx#28339] [Questions]
Right now i have to run two commands separately one is to run my Next.js project in dev mode and other is watch the workspace dependences and auto build on change

https://nx.dev/recipes/running-tasks/workspace-watching

How do I run both commands as one instead of running 2 commands everytime in different terminals ?
```sh
npx run sa:dev
npx nx watch --projects=sa --includeDependentProjects -- nx run-many -t dev -p \$NX_PROJECT_NAME --exclude=sa
```

**Comment by pankyriazo** [comment]
Hi there 👋 ! 
I came across this question while randomly searching for something else.
Here’s what’s worked for me in my projects:

```
{
  "name": "sa",
  "targets": {
    "dev": {
      "executor": "nx:run-commands",
      "options": {
        "command": "npx nx run-many --projects=sa --targets=dev:sa,dev:dependencies"
      }
    },
    "dev:sa": {
      "executor": "nx:run-commands",
      "options": {
        "command": "npx run sa:dev"
      }
    },
    "dev:dependencies": {
      "executor": "nx:run-commands",
      "options": {
        "command": "npx nx watch --projects=sa --includeDependentProjects -- nx run-many -t build -p \\$NX_PROJECT_NAME --exclude=sa"
      }
    }
  }
}
```

Then i run `npx nx run sa:dev`.

Hope this helps!

**Comment by ImBIOS** [comment]
For some folks out there who are using extensions of Nx, for example `@nx/next`, there is no need to build the library.

> There is no need to build the library prior to using it. When you update your library, the Next.js application will automatically pick up the changes.

source: https://nx.dev/nx-api/next#using-an-nx-library-in-your-application

**Reply by Herber230** [reply]
Maybe there is an error in the NX 22 because it is not working even with a completely new workspace.

**Reply by jtw-r** [reply]
The link is broken since NX keeps reshuffling their website every couple of months 🙄. Here's an updated one: https://nx.dev/docs/technologies/react/next/introduction#using-an-nx-library-in-your-application

**Comment by khanakia** [comment]
Not using NX anymore, migrated to Turbo. So closing

**Reply by Mush-A** [reply]
interesting, can you tell me why you would do that?
